docs/guides/advanced-use-case/cross_cluster_backup.md

```diff
@@ -0,0 +1,582 @@
+
+```yaml
+apiVersion: stash.appscode.com/v1alpha1
+kind: Repository
+metadata:
+  name: gcs-repo
+  namespace: demo
+spec:
+  backend:
+    gcs:
+      bucket: stash-testing
+      prefix: /cross-cluster/deployment/sample-deployment
+    storageSecretName: gcs-secret
+```
+
+Let's create the Repository we have shown above,
+```bash
+$ kubectl apply -f https://github.com/stashed/docs/raw/{{< param "info.version" >}}/docs/examples/guides/advanced-use-case/cross-cluster-backup/repository_prod.yaml
+repository.stash.appscode.com/gcs-repo created
+```
+
+Now, we are ready to restore our sample data from this specified backend.
+
+## Restore
+
+**Create RestoreSession:**
+
+Now, we need to create a `RestoreSession` crd targeting the `stash-recovered` Deployment to restore the backed-up data inside it.
+
+Below is the YAML of the `RestoreSesion` crd that we are going to create,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: RestoreSession
+metadata:
+  name: deployment-restore
+  namespace: demo
+spec:
+  repository:
+    name: gcs-repo
+  target: # target indicates where the recovered data will be stored
+    ref:
+      apiVersion: apps/v1
+      kind: Deployment
+      name: stash-recovered
+    volumeMounts:
+    - name:  source-data
+      mountPath:  /source/data
+    rules:
+    - paths:
+      - /source/data/
+```
+
+Here,
+
+- `spec.repository.name` specifies the name of the Repository.
+- `spec.target.ref` refers to the target workload where the recovered data will be stored.
+- `spec.target.volumeMounts` specifies a list of volumes and their mountPath where the data will be restored.
+  - `mountPath` must be the same mountPath as the original volume because Stash stores the absolute path of the backed-up files. If you use a different mountPath for the restored volume the backed up files will not be restored into your desired volume.
+
+Let's create the `RestoreSession` crd we have shown above,
+
+```bash
+$ kubectl apply -f https://github.com/stashed/docs/raw/{{< param "info.version" >}}/docs/examples/guides/advanced-use-case/cross-cluster-backup/restoresession_staging.yaml
+restoresession.stash.appscode.com/deployment-restore created
+```
+
+Once, you have created the `RestoreSession` crd, Stash will inject `init-container` into the `stash-recovered` Deployment. The Deployment will restart and the `init-container` will restore the desired data on start-up.
+
+### Verify Restore
+
+**Wait for RestoreSession to Succeeded:**
+
+Run the following command to watch the RestoreSession phase,
+
+```bash
+$ kubectl get restoresession -n demo -w
+
+NAME                 REPOSITORY   PHASE       DURATION   AGE
+deployment-restore   gcs-repo     Succeeded   10s        35s
+```
+
+So, we can see from the output of the above command that the restore process succeeded.
+
+
+**Verify Restored Data:**
+
+In this section, we are going to verify that the desired data has been restored successfully.
```

Comment:
In this section, we are going to verify whether the desired data has been restored successfully or not.